git_repo/services/ext/gerrit.py

```diff
@@ -0,0 +1,163 @@
+        branches = self.project_client.get_branches(project['name'])
+        for branch in branches:
+            if branch['ref'] == 'HEAD':
+                return branch['revision']
+
+    def is_repository_empty(self, project):
+        # There is no way to find out if repository is empty, so always return False
+        return False
+
+    def get_parent_project_url(self, namespace, repo, rw=True):
+        # Gerrit parent project concept is quite different from other services,
+        # so it is better to always return None here
+        return None
+
+    def request_create(self, onto_user, onto_repo, from_branch, onto_branch=None, title=None, description=None, auto_slug=False, edit=None):
+        from_branch = from_branch or self.repository.active_branch.name
+        onto_branch = onto_branch or 'HEAD:refs/for/' + from_branch
+        remote = self.repository.remote(self.name)
+        info, lines = self.push(remote, onto_branch)
+        new_changes = []
+        new_changes_lines = False
+        for line in lines:
+            if line.startswith('remote:'):
+                line = line[len('remote:'):].strip()
+
+                if 'New Changes' in line:
+                    new_changes_lines = True
+
+                if new_changes_lines and self.fqdn in line:
+                    url = line.split(' ')[0]
+                    new_changes.append(url)
+
+        if len(new_changes) > 0:
+            yield '{}'
+            yield ['Created new review request of `{local}` onto `{project}:{remote}`'.format(
+                local = from_branch,
+                project = '/'.join([onto_user, onto_repo]),
+                remote = onto_branch
+            )]
+            for url in new_changes:
+                yield ['with changeset {} available at {}'.format(url, url.split('/')[-1])]
+        else:
+            yield '{}'
+            yield ['Review request of `{local}` was not created'.format(
+                local = from_branch
+            )]
+            for element in info:
+                yield ['{} -> {}: {}'.format(element.local_ref, element.remote_ref_string, element.summary)]
+
+    def request_fetch(self, user, repo, request, pull=False, force=False):
+        if 'refs/changes/' not in request:
+            if '/' in request:
+                change_id, patch_set = request.split('/')
+            else:
+                change_id = request
+                change = self.change_client.get_all(['change: {}'.format(change_id)], ['CURRENT_REVISION'])[0]
+                current_patchset = change['revisions'][change['current_revision']]
+                patch_set = current_patchset['_number']
+
+            if change_id[0] == 'I':
+                change_id = str(self.change_client.get_by_id(request)['_number'])
+
+            request = 'refs/changes/{}/{}/{}'.format(change_id[-2:], change_id, patch_set)
+        else:
+            change_id = request.split('/')[3]
+
+        remote = self.repository.remote(self.name)
```

Comment:
what if the gerrit `remote` does not exists?

maybe handling an error and issue a warning guiding the user to do something like: `git-repo gerrit add <slug>` (or doing it directly) could be a good idea ;)

Reply:
Done. Could you check please and tell if message is okay. If you have some ideas, we may change it to something better if needed.